core/drand_beacon.go

```diff
@@ -177,11 +177,18 @@ func (bp *BeaconProcess) WaitDKG() (*key.Group, error) {
 	// filter the nodes that are not present in the target group
 	var qualNodes []*key.Node
 	for _, node := range bp.dkgInfo.target.Nodes {
+		found := false
 		for _, qualNode := range res.Result.QUAL {
 			if qualNode.Index == node.Index {
 				qualNodes = append(qualNodes, node)
+				found = true
+				break
 			}
 		}
+
+		if !found {
+			bp.log.Debugw("disqualified node during DKG", "node", node)
+		}
 	}
```

Comment:
this code hurts my soul, but the number of nodes is so low it doesn't seem worth refactoring to use a map

Reply:
as-is though, I think you could remove `found` and just do
```
for i, qualNode := range res.Result.QUAL {
	if qualNode.Index == node.Index {
		qualNodes = append(qualNodes, node)
		break
	} else if i == len(res.Result.QUAL) {
		bp.log.Debugw("disqualified node during DKG", "node", node)
	}
}
```